airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/FileBuffer.java

```diff
@@ -25,14 +25,33 @@ public class FileBuffer implements BufferStorage {
   // "To optimize the number of parallel operations for a load,
   // we recommend aiming to produce data files roughly 100-250 MB (or larger) in size compressed."
   public static final long MAX_PER_STREAM_BUFFER_SIZE_BYTES = 200 * 1024 * 1024; // mb
-  // Other than the per-file size limit, we also limit the total size (which would limit how many
-  // concurrent streams we can buffer simultaneously too)
-  // Since this class is storing data on disk, the buffer size limits below are tied to the
-  // necessary disk storage space.
+  /*
+   * Other than the per-file size limit, we also limit the total size (which would limit how many
+   * concurrent streams we can buffer simultaneously too) Since this class is storing data on disk,
+   * the buffer size limits below are tied to the necessary disk storage space.
+   */
   public static final long MAX_TOTAL_BUFFER_SIZE_BYTES = 1024 * 1024 * 1024; // mb
-  // we limit number of stream being buffered simultaneously anyway (limit how many files are
-  // stored/open for writing)
-  public static final int DEFAULT_MAX_CONCURRENT_STREAM_IN_BUFFER = 10;
+  /*
+   * We limit number of stream being buffered simultaneously anyway (limit how many files are
+   * stored/open for writing)
+   *
+   * Note: This value can be tuned to increase performance with the tradeoff of increased memory usage
+   * (~31 MB per buffer). See {@link StreamTransferManager}
+   *
+   * For connections with interleaved data (e.g. Change Data Capture), having less buffers than the
+   * number of streams being synced will cause buffer thrashing where buffers will need to be flushed
+   * before another stream's buffer can be created. Increasing the default max will reduce likelihood
+   * of thrashing but not entirely eliminate unless number of buffers equals streams to be synced
+   */
+  public static final int DEFAULT_MAX_CONCURRENT_STREAM_IN_BUFFER = 15;
```

Comment:
We are changing the default and minimum from 10 to 15. This raises two questions:
 - How this will affect existing connections? 
 - How this will affect non-cloud users, who may be running this connector in containers with less than 1GB RAM?

Reply:
* The likelihood of this affecting existing connections will be low since the minimum memory needed for all the default buffers will be 465 MB (15 * 31 MB) which is strictly less than the 1 GB `MAX_TOTAL_BUFFER_SIZE_BYTES` 
* Furthermore, if a customer had low memory availability then they would have run into issues earlier since our `MAX_TOTAL_BUFFER_SIZE_BYTES` was 1 GB before a flush. To not run into issues prior, users would have had to tune the original `DEFAULT_MAX_CONCURRENT_STREAM_IN_BUFFER` or lower the `MAX_TOTAL_BUFFER_SIZE_BYTES` to the amount of memory available for the JVM

It is reasonable to have within the change-log to mark this change as potentially breaking to users with memory less than 500 MB

Reply:
AFAIK, we used to run containers with 500MB by default and recently switched to 1GB. Why not leave minimum as `10` instead?

Reply:
The `MAX_TOTAL_BUFFER_SIZE_BYTES` has been 1 GB since March so most of this decision is based on that value. Didn't realize that you already had a comment about this [here](https://github.com/airbytehq/airbyte/pull/15707#discussion_r1040178169) in the community PR but it makes sense to reduce plausibilities of breakages. It begs the question of how we communicate better with our community about these increases. It seems like allowing it to be configurable is one option with proper warnings

Based on your comment, it seems better to just have the default maintain `10` with the option to configure for the user